src/components/EditableTypography/EditableTypography.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { ReactNode, forwardRef, useEffect, useRef, useState } from "react";
+import cx from "classnames";
+import { useMergeRefs } from "../../hooks";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import VibeComponent from "../../types/VibeComponent";
+import styles from "./EditableTypography.module.scss";
+import HiddenInputPlaceholder from "./HiddenInputPlaceholder";
+import { keyCodes } from "../../constants";
+import useClickableProps from "../../hooks/useClickableProps/useClickableProps";
+
+export interface EditableTypographyImplementationProps {
+  /** Value of the text */
+  value: string;
+  /** Will be called whenever the current value changes to a non-empty value */
+  onChange?: (value: string) => void;
+  /** Disables editing mode - component will be just a <Heading /> */
+  readOnly?: boolean;
+  /** Shown in edit mode when the text value is empty */
+  placeholder?: string;
+  /** ARIA Label */
+  ariaLabel?: string;
+}
+
+interface EditableTypographyProps extends VibeComponentProps, EditableTypographyImplementationProps {
```

Comment:
These props are only require for the composition and should not be exposed from the composed components, this is why the types are separated